### DISCUSSION TITLE
title of csv exports

### DISCUSSION THREAD

**Discussion by mcdangermark** [dbeaver/dbeaver#14676] [Q&A]
Hi, 

I am working in a new role where I have learned about as much SQL, DAX and Mquery as needed to pull csv files from my organizations database using DBeaver, plug them into Power BI and manipulate as needed. However, I am limited in my knowledge of how to troubleshoot issues when that flow is interrupted :-( 

I have recently begun trying to expand on my lexicon of SQL in order to pare down the size of my csv exports (going from SELECT * to more specific statements). In doing so, I am encountering an infuriating issue, where some of my CSVs will export as the name of the original table as intended (e.g. schema.table), and others will export titled as the name of my SQL statement ( SELECT yada, yada, yada, yada, yada). I cannot for the life of me determine how to get them to export as I would like them to consistently. Any help would be much appreciated.

**Comment by sknextcode** [comment]
Hi,

when exporting yur data to csv with DBeaver, you should be able to tell the wizard in which format the file should be named and saved. In the output step of the export wizard, the option "file name pattern" should be located right beneath the directory selection.
If you simply enter `${schema}.${table}`in there, you should get what you want.
However, `${schema}` seems to not work with query result export at the moment (v. 21.2.5), only when exporting a whole table.
Yo can also look it up here: https://dbeaver.com/docs/wiki/Data-transfer/
In step 4 there is a file name pattern of `${table}_${timestamp}` for reference.

Hope you find that information helpful.
